tests/test_workflows.py

```diff
@@ -119,4 +119,37 @@ async def event_workflow(c: Context):
         assert s.w_output == "Hello from outside! B A"
 
 
-# TODO: test concurrent
+@async_test
+async def test_utils_methods(t: WorkflowTest):
+    @workflow()
+    async def utils_workflow(c: Context):
+        current_time = await c.utils.now()
+        rand_value = await c.utils.random(1, 10)
+        unique_id = await c.utils.uuid4()
+        return str(current_time), rand_value, str(unique_id)
+
+    backends = [
+        FSBackend("./logs"),
+        RedisBackend(host="localhost", port=6380, password="password"),
+    ]
+
+    for backend in backends:
+        s = await t.step(backend).exec_workflow(utils_workflow)
+
+        current_time, rand_value, unique_id = s.w_output
+
+        assert len(s.w_records.events) > 0
+
+        saved_value_now, saved_value_rand, saved_value_uuid = [
+            json.loads(event.save.value)
+            for event in s.w_records.events
+            if event.is_set("save")
+        ]
```

Comment:
Not quite.. here you are returning the saved value then compared it against itself.
As if to say..
```py
def foo():
    a = "thing"
    return a
assert foo() == "thing"
```

Reply:
Also, you should trrigger a replay with the help of `await c.sleep(timedelta(seconds=1))` or similar before the return.